x/tokenfactory/keeper/createdenom.go

```diff
@@ -77,7 +78,12 @@ func (k Keeper) chargeForCreateDenom(ctx sdk.Context, creatorAddr string, subden
 	if err != nil {
 		return err
 	}
-	if len(creationFee) > 0 {
+	if creationFee != nil {
+		for _, coin := range creationFee {
+			if !k.bankKeeper.HasBalance(ctx, accAddr, coin) {
+				return sdkerrors.ErrInsufficientFunds
+			}
+		}
```

Comment:
this was added after I found that if the creation fee was multiple denoms, it was possible for a user to be charged if they had one of the two required denoms and still not get to create a tokenfactory token

Reply:
Wait what? FundCommunityPool should call SendCoins which does this check?

Did we have a test for this showing the problem, this seems like something very concerning if true.

Reply:
We didn't have a test for this, and it only showed its head when I added the `account doesn't have enough to pay for denom creation fee` test case. If you revert just this change and run the test case with prints you will see the following:

```
preCreateBalance 100000000uion,1000000000uosmo 
postCreateBalance 50000000uion,1000000000uosmo 
```

We had enough uion so it deducted that, but we didnt have enough uosmo and stops there.

Reply:
This is an error in the test. 

The way reversion is supposed to work, is that all tx state changes outside of the ante handler get reverted if the message returns an error. So the test is wrong here, the code was right.

The test was for the keeper function, which is the incorrect layer to test reversion logic.

Reply:
Ah I see, my apologies for raising a false flag! Will revert this